File: scripts/linear-get-issue.py
"""Get full details of Linear issues by identifier."""
import urllib.request, json, sys

import os
TOKEN = os.environ.get("LINEAR_API_KEY", "")

def gql(query, variables=None):
    payload = json.dumps({"query": query, "variables": variables or {}}).encode()
    req = urllib.request.Request(
        "https://api.linear.app/graphql",
        data=payload,
        headers={"Content-Type": "application/json", "Authorization": TOKEN}
    )
    with urllib.request.urlopen(req) as r:
        return json.loads(r.read())

for identifier in sys.argv[1:]:
    q = """
    query($id: String!) {
      issue(id: $id) {
        id identifier title priority
        description
        state { name type }
        labels { nodes { name } }
      }
    }
    """
    result = gql(q, {"id": identifier})
    issue = result["data"]["issue"]
    print(f"\n{'='*60}")
    print(f"{issue['identifier']}: {issue['title']}")
    print(f"State: {issue['state']['name']} | Priority: {issue['priority']}")
    labels = [n["name"] for n in issue["labels"]["nodes"]]
    if labels:
        print(f"Labels: {', '.join(labels)}")
    desc = (issue.get("description") or "").encode("ascii", "replace").decode()
    print(f"\nDescription:\n{desc[:2000]}")

File: scripts/linear-get-issue.mjs
/**
 * Fetch full details of a specific Linear issue.
 * Run: LINEAR_API_KEY=lin_api_xxx node scripts/linear-get-issue.mjs UNI-1513
 */

import { linearGql } from './linear-update.mjs';

const identifier = process.argv[2];
if (!identifier) { console.error('Usage: node linear-get-issue.mjs <IDENTIFIER>'); process.exit(1); }

const data = await linearGql(`{
  issue(id: "${identifier}") {
    id
    identifier
    title
    priority
    description
    state { id name type }
    team { key name states { nodes { id name type } } }
    children { nodes { identifier title state { name } } }
  }
}`);

const i = data.issue;
console.log(`\n=== ${i.identifier}: ${i.title} ===`);
console.log(`State: ${i.state.name} | Team: ${i.team.name}`);
console.log(`\nDescription:\n${i.description || '(none)'}`);
if (i.children?.nodes?.length) {
  console.log(`\nSub-issues:`);
  i.children.nodes.forEach(c => console.log(`  [${c.identifier}] ${c.title} — ${c.state.name}`));
}
